src/Aspire.ProjectTemplates/templates/aspire-starter/.template.config/template.json

```diff
@@ -344,27 +344,49 @@
       "datatype": "bool",
       "defaultValue": "false",
       "description": "Whether to turn off HTTPS."
+    },
+    "GeneratedClassNamePrefix": {
+      "type": "generated",
+      "generator": "regex",
+      "datatype": "string",
+      "replaces": "GeneratedClassNamePrefix",
+      "parameters": {
+        "source": "name",
+        "steps": [
+          {
+            "regex": "[\\.\\- ]",
```

Comment:
The template engine uses this for the class name: 
https://github.com/dotnet/templating/blob/a325ffa18edd1590f9b340cf83d51d8eb567ebdc/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ValueForms/DefaultSafeNameValueFormFactory.cs#L19

.. which seems to be better as it catches more cases - `(((?<=\.)|^)(?=\d)|\W)`.

Reply:
This patch fixes it for both template generated names, and our own - https://gist.github.com/radical/5c746337ecf8136fa240f4fd1ea1086f 

Tested with `"aspire & foo-bar"`.

Reply:
It needs to match what we do in our targets though